calico-node(v3.0.1) panic when in canal setup using etcd datastore

When I was testing out Canal manifests for #1642 (actually using the v3.0.1 release, not master) I hit the following panic when installing fresh or doing an upgrade and using an etcd datastore.

```
core@k8s-master ~ $ kubectl logs -n kube-system canal-node-z9qwj -c calico-node
2018-01-26 21:30:41.359 [INFO][7] startup.go 187: Early log level set to info
2018-01-26 21:30:41.359 [INFO][7] startup.go 198: NODENAME environment not specified - check HOSTNAME
2018-01-26 21:30:41.360 [INFO][7] startup.go 100: Skipping datastore connection test
2018-01-26 21:30:41.366 [INFO][7] startup.go 153: Using node name: k8s-node-01
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x112af56]

goroutine 1 [running]:
main.removeHostTunnelAddr(0x1c8ef60, 0xc42000e1e8, 0x1c95f20, 0xc42011cd80, 0xc42000a149, 0xb)
        /go/src/github.com/projectcalico/calico/calico_node/allocateipip/allocate_ipip_addr.go:112 +0xf6
main.main()
        /go/src/github.com/projectcalico/calico/calico_node/allocateipip/allocate_ipip_addr.go:56 +0x1da
Calico node failed to start
```

## Expected Behavior
calico-node works and applies policy

## Current Behavior
calico-node panics

## Possible Solution
Fix it

## Steps to Reproduce (for bugs)
1. Setup according to the docs in #1642
2. Check the calico-node logs after applying the Canal manifest

## Context
Found while trying to test out updated Canal manifest.

## Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->
* Calico version
* Orchestrator version (e.g. kubernetes, mesos, rkt):
* Operating System and version:
* Link to your project (optional):

--- a/calico_node/allocateipip/allocate_ipip_addr.go
+++ b/calico_node/allocateipip/allocate_ipip_addr.go
@@ -74,7 +74,7 @@ func ensureHostTunnelAddress(ctx context.Context, c client.Interface, nodename s
 		log.WithError(err).Fatalf("Unable to retrieve IPIP tunnel address. Error getting node '%s'", nodename)
 	}
 
-	if node.Spec.BGP.IPv4IPIPTunnelAddr == "" {
+	if node.Spec.BGP == nil || node.Spec.BGP.IPv4IPIPTunnelAddr == "" {
 		// The IPIP tunnel has no IP address assigned, assign one.
 		log.Debug("IPIP tunnel is not assigned - assign IP")
 		assignHostTunnelAddr(ctx, c, nodename, ipipCidrs)
@@ -113,7 +113,7 @@ func removeHostTunnelAddr(ctx context.Context, c client.Interface, nodename stri
 			log.WithError(err).Fatalf("Unable to retrieve IPIP tunnel address for cleanup. Error getting node '%s'", nodename)
 		}
 
-		if node.Spec.BGP.IPv4IPIPTunnelAddr == "" {
+		if node.Spec.BGP == nil || node.Spec.BGP.IPv4IPIPTunnelAddr == "" {
 			log.Debug("No IPIP tunnel address assigned, and not required")
 			return
 		}
@@ -174,6 +174,9 @@ func assignHostTunnelAddr(ctx context.Context, c client.Interface, nodename stri
 			log.WithError(err).Fatalf("Unable to retrieve IPIP tunnel address for cleanup. Error getting node '%s'", nodename)
 		}
 
+		if node.Spec.BGP == nil {
+			node.Spec.BGP = &api.NodeBGPSpec{}
+		}
 		node.Spec.BGP.IPv4IPIPTunnelAddr = ipv4Addrs[0].String()
 
 		_, updateError = c.Nodes().Update(ctx, node, options.SetOptions{})

